### ISSUE TITLE
undefined method `type' for nil:NilClass (NoMethodError) when calling `reek`

### ISSUE THREAD

**Issue by letz** [troessner/reek#708]
When I try to run the command `reek` or `reek .` it gives me the following error

```
undefined method `type' for nil:NilClass (NoMethodError)
```

Any ideas?

Stack trace:
```
$ reek
/Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/ast/sexp_extensions.rb:199:in `attr_with_writable_flag?': undefined method `type' for nil:NilClass (NoMethodError)
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/ast/sexp_extensions.rb:193:in `attribute_writer?'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:95:in `process_send'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:137:in `process_begin'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:46:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:70:in `block in process_def'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:200:in `block in inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:213:in `push'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:199:in `inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:68:in `process_def'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:53:in `block in process_module'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:200:in `block in inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:213:in `push'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:199:in `inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:52:in `process_module'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:137:in `process_begin'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:83:in `block in process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:82:in `process_default'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:53:in `block in process_module'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:200:in `block in inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:213:in `push'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:199:in `inside_new_context'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:52:in `process_module'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:44:in `process'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:38:in `result'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/tree_walker.rb:27:in `walk'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/examiner.rb:87:in `run'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/examiner.rb:42:in `initialize'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/cli/reek_command.rb:13:in `new'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/cli/reek_command.rb:13:in `block in execute'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/cli/reek_command.rb:12:in `each'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/cli/reek_command.rb:12:in `execute'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/lib/reek/cli/application.rb:33:in `execute'
	from /Users/letz/.rvm/gems/ruby-2.2.3/gems/reek-3.4.0/bin/reek:12:in `<top (required)>'
	from /Users/letz/.rvm/gems/ruby-2.2.3/bin/reek:23:in `load'
	from /Users/letz/.rvm/gems/ruby-2.2.3/bin/reek:23:in `<main>'
	from /Users/letz/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `eval'
	from /Users/letz/.rvm/gems/ruby-2.2.3/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by chastell** [comment]
Can you try isolating the exact code that blows up? [It seems](https://github.com/troessner/reek/blob/v3.4.0/lib/reek/ast/sexp_extensions.rb#L198-L200) like an `attr` call with no arguments…

**Comment by letz** [comment]
I just installed the gem and typed `reek` in the terminal

**Comment by andyw8** [comment]
Without seeing your code which is being parsed it's very difficult to know what's going wrong.

Can you try running reek on each directory in your project individually (e.g. `reek lib`), and then, if possible, narrow down the individual file which causes the parsing error?

**Comment by chastell** [comment]
I think I might’ve replicated this with

```ruby
class Foo
  attr
end
```

This is, surprisingly, valid Ruby code that blows up (with the above) when reek tries to parse it.

**Comment by mvz** [comment]
> This is, surprisingly, valid Ruby code

What does it do?

**Comment by troessner** [comment]
>> What does it do?

I'd love to know that as well :laughing:

**Comment by chastell** [comment]
I would assume ‘nothing much’, but we should support it nonetheless (as in, ‘at least not blow up’). :)

**Comment by mvz** [comment]
> at least not blow up

Agreed :laughing:.

**Comment by letz** [comment]
thanks!

**Comment by chastell** [comment]
This is now fixed in the freshly-released reek 3.4.1 – thanks for the bug report!
